homeassistant/components/velux/config_flow.py

```diff
@@ -0,0 +1,70 @@
+"""Config flow for Velux integration."""
+from typing import Any
+
+from pyvlx import PyVLX, PyVLXException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import _LOGGER, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for velux."""
+
+    VERSION: int = 1
+
+    @staticmethod
+    async def test_connection(host: str, password: str):
+        """Test the connection to Velux."""
+        pyvlx = PyVLX(host=host, password=password)
+        await pyvlx.connect()
+        await pyvlx.disconnect()
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import a config entry."""
+        self._async_abort_entries_match({CONF_HOST: config[CONF_HOST]})
+        return self.async_create_entry(
+            title=config[CONF_HOST],
+            data=config,
+        )
+
+    async def async_step_user(
```

Comment:
Is there some kind of serial number or anything we can get from the device so we set the config flow unique_id?

Reply:
Hostname is quite unique. Extract from API spec: 
```
mDNS responds with 2 instance names: "klf-200_nXXXX._http._tcp.local" and "klf200_nYYYY._http._tcp.local" where XXXX and YYYY are the last 4 hexadecimal digits of LAN and WiFi MAC addresses. 
```

Reply:
How can I get the hostname? Is there a method in pyvlx? If so, I have been unable to find it.

Reply:
I asked someone about it, will update when I have response

Reply:
Unfortunately the hostname will not be provided from the API. It needs to be gathered from zeroconf. Below is an example:
```
    async def async_step_zeroconf(
        self, discovery_info: ZeroconfServiceInfo
    ) -> FlowResult:
        """Handle discovery by zeroconf."""
        hostname = discovery_info.hostname.replace(".local.", "")
        await self.async_set_unique_id(hostname)
        self._abort_if_unique_id_configured(updates={CONF_HOST: discovery_info.host})

        # Check if config_entry exists already without unigue_id configured.
        for entry in self.hass.config_entries.async_entries(DOMAIN):
            if entry.data[CONF_HOST] == discovery_info.host and entry.unique_id is None:
                entry.unique_id = hostname
                return self.async_abort(reason="already_configured")
```
in order to get this running, you need to extend the manifest: 
```
  "zeroconf": [
    {"type":"_http._tcp.local.","name":"velux_klf_lan*"}
    ],
```

Reply:
@joostlek Are there any news? The solution suggested by @pawlizio does not work for me to read the hostname in the normal config flow, because it depends on zeroconf data.

Reply:
I just asked around. For now we should not set the unique_id.

When adding zeroconf, we should add zeroconf support to the pyvlx library, which will be used to get the identifier we will use to set the entries that have no unique_id yet.